src/loaders/VolumeLoaderUtils.ts

```diff
@@ -76,3 +78,25 @@ export function estimateLevelForAtlas(spatialDimsZYX: number[][], maxAtlasEdge =
   }
   return levelToLoad;
 }
+
+// currently everything needed can come from the imageInfo
+// but in the future each IVolumeLoader could have a completely separate implementation.
+export function buildDefaultMetadata(imageInfo: ImageInfo): Record<string, unknown> {
+  const metadata = {};
+  metadata["Dimensions"] = { x: imageInfo.tile_width, y: imageInfo.tile_height, z: imageInfo.tiles };
```

Comment:
This could also be expressed as a large object literal, e.g.
```
metadata = {
  "Original dimensions": { x: imageInfo.width, y: imageInfo.height, z: imageInfo.tiles },
  ...
}
```
...though I don't have strong feelings about which is preferable.

Reply:
Same here, don't really care.  Imperative code is more ingrained in me than declarative, especially when there might be some complexity to arrive at the values being set.

Reply:
My instinct would have been to go declarative, but given that "User data" may or may not be added to the object at all, I think it makes sense the way it is